### PR TITLE
lint: unused local declarations

### PR DESCRIPTION
**Pull request by nektro**
found with https://github.com/nektro/ziglint and manually fixed

### CODE REVIEW

File: lib/std/c/freebsd.zig [hunk 1]
```diff
@@ -944,8 +944,6 @@ pub const winsize = extern struct {
     ws_ypixel: u16,
 };
 
-const NSIG = 32;
```
Comment (andrewrk) [inline lib/std/c/freebsd.zig:947]: The fix for this one is to make it `pub`. Same with all the other instances.

File: lib/std/crypto/siphash.zig [hunk 2]
```diff
@@ -46,8 +46,6 @@ fn SipHashStateless(comptime T: type, comptime c_rounds: usize, comptime d_round
 
     return struct {
         const Self = @This();
-        const block_length = 64;
-        const digest_length = 64;
```
Comment (andrewrk) [inline lib/std/crypto/siphash.zig:50]: These are observable from outside siphash.zig. They should be `pub`.

File: lib/std/start.zig [hunk 3]
```diff
@@ -1,12 +1,12 @@
 // This file is included in the compilation unit when exporting an executable.
+//
```
Comment (andrewrk) [inline lib/std/start.zig:2]: stray `//`

File: lib/std/zig/system/linux.zig [hunk 4]
```diff
@@ -268,6 +267,7 @@ const ArmCpuinfoImpl = struct {
                 0x43 => &Cavium,
                 0x46 => &Fujitsu,
                 0x48 => &HiSilicon,
+                0x4e => &Nvidia,
```
Comment (andrewrk) [inline lib/std/zig/system/linux.zig:270]: Nice fix

File: test/behavior/array.zig [hunk 5]
```diff
@@ -58,9 +57,6 @@ test "array dot len const expr" {
     });
 }
 
-const ArrayDotLenConstExpr = struct {
```
Comment (andrewrk) [inline test/behavior/array.zig:61]: please check the git history of this and make sure a test that referenced this decl was not accidentally deleted

File: test/behavior/sizeof_and_typeof.zig [hunk 6]
```diff
@@ -1,10 +1,10 @@
 const std = @import("std");
 const expect = std.testing.expect;
-const expectEqual = std.testing.expectEqual;
 
 test "@sizeOf and @TypeOf" {
     const y: @TypeOf(x) = 120;
     try expect(@sizeOf(@TypeOf(y)) == 2);
+    try expect(@sizeOf(@TypeOf(z)) == 2);
```
Comment (andrewrk) [inline test/behavior/sizeof_and_typeof.zig:7]:
```suggestion
    try expect(@TypeOf(z) == u16);
```

File: test/behavior/typename.zig [hunk 7]
```diff
@@ -1,5 +1,5 @@
+//
```
Comment (andrewrk) [inline test/behavior/typename.zig:1]: stray line

**Review by andrewrk — CHANGES_REQUESTED** [review]
Thanks for the PR. Left some requested changes.

File: test/behavior/typename.zig (hunk 7 above)
Comment (nektro) [reply test/behavior/typename.zig:1]: needed for source sensitive test

File: lib/std/start.zig (hunk 3 above)
Comment (nektro) [reply lib/std/start.zig:2]: needed for source sensitive test

File: lib/std/crypto/siphash.zig (hunk 2 above)
Comment (nektro) [reply lib/std/crypto/siphash.zig:50]: https://discord.com/channels/697519923273924658/713755801583616100/929803897252511794

Comment (andrewrk) [reply lib/std/crypto/siphash.zig:50]: Can you post the content here?

File: lib/std/os/linux/start_pie.zig [hunk 8]
```diff
@@ -16,6 +15,7 @@ const R_RELATIVE = switch (builtin.cpu.arch) {
     .arm => R_ARM_RELATIVE,
     .aarch64 => R_AARCH64_RELATIVE,
     .riscv64 => R_RISCV_RELATIVE,
+    .sparc => R_SPARC_RELATIVE,
```
Comment (nektro) [inline lib/std/os/linux/start_pie.zig:18]: should this be `.sparc` or `.sparcv9`

File: lib/std/start.zig (hunk 3 above)
Comment (ifreund) [reply lib/std/start.zig:2]: > needed for source sensitive test

what does this mean?

Comment (andrewrk) [reply lib/std/start.zig:2]: Ah, the stack trace test checks lines and columns and unfortunately that means the test case breaks whenever lines are added or removed from start.zig.

The annoying but proper fix is to adjust the test case expected line number. Or just don't mess with this file in this PR.

File: test/behavior/typename.zig (hunk 7 above)
Comment (andrewrk) [reply test/behavior/typename.zig:1]: not sure what you're talking about on this one

File: lib/std/os/linux/start_pie.zig (hunk 8 above)
Comment (andrewrk) [reply lib/std/os/linux/start_pie.zig:18]: definitely at least `.sparc`. looks like also `.sparcv9`

File: test/behavior/typename.zig (hunk 7 above)
Comment (Hejsil) [reply test/behavior/typename.zig:1]: I think this refers to the fact that these files source-location sensitive, as some of these type names contains source lines and columns in the names.

File: test/behavior/array.zig (hunk 5 above)
Comment (nektro) [reply test/behavior/array.zig:61]: <https://github.com/ziglang/zig/commit/581edd643> first added in its modern form here
<https://github.com/ziglang/zig/commit/0cd361219> moved to `_stage1.zig` here
<https://github.com/ziglang/zig/commit/6d6cf5984> implemented here

Comment (nektro) [reply test/behavior/array.zig:61]: asserting that `y: [some_array.len]u8` properly turned to `y: [4]u8` is likely what it was meant to test, but never added

Comment (nektro) [reply test/behavior/array.zig:61]: <https://github.com/ziglang/zig/commit/6b3ce918d> first added ever here
<https://github.com/ziglang/zig/commit/dab3ddab4> getting closer to modern zig

**Pull request closed without merging**